src/libcore/default.rs

```diff
@@ -115,6 +115,48 @@ pub trait Default: Sized {
     fn default() -> Self;
 }
 
+/// Equivalent to [`Default::default()`] but without repeating the word
+/// "default" twice at the call site.
+///
+/// For example:
+/// ```
+/// #![feature(default_free_fn)]
+///
+/// use std::default::default;
+///
+/// #[derive(Default)]
+/// struct AppConfig {
+///     foo: FooConfig,
+///     bar: BarConfig,
+/// }
+///
+/// #[derive(Default)]
+/// struct FooConfig {
+///     foo: i32,
+/// }
+///
+/// #[derive(Default)]
+/// struct BarConfig {
+///     bar: f32,
+///     baz: u8,
+/// }
+///
+/// fn main() {
+///     let options = AppConfig {
+///         foo: default(),
+///         bar: BarConfig {
+///             bar: 10.1,
+///             ..default()
+///         },
+///     };
+/// }
+/// ```
+#[unstable(feature = "default_free_fn", issue = "73014")]
+#[inline]
+pub fn default<T: Default>() -> T {
+    Default::default()
```

Comment:
This is a nitpick I'd mention in any PR, but I guess it's extra strange now - here, why don't we prefer using the style `T::default()`?

Reply:
I would like to stick with `Default::default()`. The documentation of the function describes it as being equivalent to `Default::default()` and this reinforces that, even if `T::default()` would also be equivalent. It would be less clear for the documentation to describe the function as being equivalent to `T::default()` because `T::default()` has type `T` while `Default::default()` has a type that is filled in by inference.

Reply:
I guess I'm here because avoiding `Default::default` is a pet peeve of mine, which in some way I must partly share with ilya-bobyr then. Since we say equivalent, any equivalent code would be fine, and I think we should prefer the style that we would like to see in Rust codebases around the ecosystem, hence `T::default()`.

If the docs say `Default::default`, maybe that could change now or in the future.

Reply:
`Default::default()` was the first thing I thought about, as this is what I was trying to replace.  If @dtolnay likes `Default::default()` more here, I would just leave it.